packages/server-side-render/src/server-side-render.js

```diff
@@ -20,126 +20,105 @@ export function rendererPath( block, attributes = null, urlQueryArgs = {} ) {
 	} );
 }
 
-export class ServerSideRender extends Component {
-	constructor( props ) {
-		super( props );
-		this.state = {
-			response: null,
-		};
-	}
+export function ServerSideRender( props ) {
+	const {
+		className,
+		EmptyResponsePlaceholder,
+		ErrorResponsePlaceholder,
+		LoadingResponsePlaceholder,
+	} = props;
 
-	componentDidMount() {
-		this.isStillMounted = true;
-		this.fetch( this.props );
-		// Only debounce once the initial fetch occurs to ensure that the first
-		// renders show data as soon as possible.
-		this.fetch = debounce( this.fetch, 500 );
-	}
+	const [ response, setResponse ] = useState( null );
+	const [ isLoading, setIsLoading ] = useState( false );
+	const isMounted = useRef( true );
 
-	componentWillUnmount() {
-		this.isStillMounted = false;
-	}
+	useEffect( () => {
+		fetchData( props );
 
-	componentDidUpdate( prevProps ) {
-		if ( ! isEqual( prevProps, this.props ) ) {
-			this.fetch( this.props );
-		}
-	}
+		fetchData = debounce( fetchData, 500 );
+		return () => {
+			isMounted.current = false;
+		};
+	}, [] );
 
-	fetch( props ) {
-		if ( ! this.isStillMounted ) {
+	useEffect( () => {
+		fetchData( props );
+	}, [ props ] );
```

Comment:
This effect will run on every render because objects in the dependency array are not deep-equality-checked, and so since they are recreated on every render, `props` will always be different because it's technically a new object every time. The only way I know of to avoid re-fetching on every render is to use the `compose` package's `usePrevious` hook and Lodash's `isEqual` function like so:

```suggestion
	const prevProps = usePrevious( props );
	useEffect( () => {
		if ( ! isEqual( prevProps, props ) ) {
			fetchData( props );
		}
	} );
```